src/bin/gpfdist/gpfdist.c

```diff
@@ -836,6 +848,37 @@ static void parse_command_line(int argc, const char* const argv[],
 			exit(1);
         }
     }
+
+	/* validate opt.I and opt.O */
+	if (opt.I || opt.O)
+	{
+		extern const char* validate_transform_opt(struct transform* trlist, const char* name);
```

Comment:
Can we include transform.h instead of this extern statement? When someday arguments of the validate_transform_opt function will be changed, a compiler won't be able to detect the error.

Reply:
You are TOTALLY right. But for some historical reasons, gpfdist developers decided to do it like so (there are many examples of it in the same file). Doing it another way is a refactoring, which I don't want to include in this patch.